packages/solana/lib/wallet/SolanaWalletProvider.ts

```diff
@@ -0,0 +1,172 @@
+        const signature = nacl.sign.detached(buffer, base58.decode(base58.encode(this._signer.secretKey)));
+
+        return Buffer.from(signature).toString('hex');
+    }
+
+    public async sendTransaction(txRequest: TransactionRequest): Promise<Transaction<any>> {
+        await this.getSigner();
+        const to = new PublicKey(txRequest.to.toString());
+
+        let instruction: TransactionInstruction;
+        // Handle ERC20 Transactions
+        if (txRequest.asset && !txRequest.asset.isNative) {
+            const contractAddress = new PublicKey(txRequest.asset.contractAddress);
+            const fromTokenAccount = await getAssociatedTokenAddress(contractAddress, this._signer.publicKey);
+
+            try {
+                await createAccount(this.chainProvider.getProvider(), this._signer, contractAddress, to);
+            } catch {}
+
+            const toTokenAccount = await getAssociatedTokenAddress(contractAddress, to);
+
+            instruction = createTransferInstruction(fromTokenAccount, toTokenAccount, this._signer.publicKey, Number(txRequest.value));
+        } else {
+            // Handle SOL Transactions
+            instruction = SystemProgram.transfer({
+                fromPubkey: this._signer.publicKey,
+                toPubkey: to,
+                lamports: txRequest.value.toNumber(),
+            });
+        }
+
+        const latestBlockhash = await this.chainProvider.getProvider().getLatestBlockhash();
+
+        const transaction = new SolTransaction({
+            recentBlockhash: latestBlockhash.blockhash,
+        }).add(instruction);
+
+        const hash = await this.chainProvider.getProvider().sendTransaction(transaction, [this._signer]);
+
+        return {
+            hash,
+            value: txRequest.value.toNumber(),
+            _raw: {},
+        };
+    }
+
+    public async sendBatchTransaction(txRequests: TransactionRequest[]): Promise<Transaction<any>[]> {
+        const result: Transaction[] = [];
+        for (const txRequest of txRequests) {
+            const tx = await this.sendTransaction(txRequest);
+            result.push(tx);
+        }
+        return result;
+    }
+
+    public async sendSweepTransaction(address: AddressType, asset: Asset): Promise<Transaction<any>> {
+        const addresses = await this.getAddresses();
+        const balance = await this.chainProvider.getBalance(addresses, [asset]);
+        return await this.sendTransaction({ to: address, value: balance[0] });
+    }
+
+    public async updateTransactionFee(_tx: string | Transaction<any>, _newFee: FeeType): Promise<Transaction<any>> {
+        throw new UnimplementedMethodError('Method not supported.');
+    }
+
+    public async getBalance(assets: Asset[]): Promise<BigNumber[]> {
+        const user = await this.getAddress();
+        return await this.chainProvider.getBalance([user], assets);
+    }
+
+    public async exportPrivateKey(): Promise<string> {
+        await this.getSigner();
```

Comment:
you should either replace `getSigner` with `setSigner` or you should use the return value from it in the function